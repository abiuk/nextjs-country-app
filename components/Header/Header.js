import React from "react";
import styled from "styled-components";
import Link from "next/link";

const Root = styled.div`
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.0975);
`;

const Container = styled.div`
  max-width: 1010px;
  padding: 26px 20px;
  margin: 0 auto;
  cursor: pointer;
`;

const Title = styled(Link)`
  font-size: 28px;
  color: inherit;
  text-decoration: none;
`;

const Header = () => (
  <Root>
    <Container>
      <Title href={{ pathname: "/" }}>Countries App</Title>
    </Container>
  </Root>
);

export default Header;
