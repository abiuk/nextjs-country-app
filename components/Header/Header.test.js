import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Header from "./Header";

describe("Header", () => {
  it("renders successfully", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be shown", () => {
    const wrapper = mount(<Header />);
    expect(wrapper).not.toBeNull();
  });

  it("should show correct text", () => {
    const wrapper = mount(<Header />);
    expect(wrapper.text().includes("Countries App")).toBe(true);
  });
});
