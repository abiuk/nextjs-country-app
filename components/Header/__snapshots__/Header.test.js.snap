// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render successfully 1`] = `
<div
  className="Header__Root-sc-1p0oksi-0 bhRYIp"
>
  <div
    className="Header__Container-sc-1p0oksi-1 jlfTsF"
  >
    <div
      className="Header__ImageWrapper-sc-1p0oksi-2 iFOBbh"
    >
      <img
        alt="santa hat"
        src="/001-santa-hat.png"
      />
    </div>
    <a
      className="Header__Title-sc-1p0oksi-3 kmBZjr"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Countries App
    </a>
  </div>
</div>
`;
